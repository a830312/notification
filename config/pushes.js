export default {
  form: {
    formName: 'notification',
    inputs: {
      title: {
        label: 'Notification Title',
        name: 'title',
        type: 'text'
      },
      body: {
        label: 'Notification Body',
        name: 'body',
        type: 'text'
      },
      type: {
        name: 'type',
        type: 'hidden'
      }
    },
    action: '/pushes',
    method: 'post',
    submit: 'Push'
  }
}