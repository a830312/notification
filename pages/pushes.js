import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { isFunction as _isFunction } from 'lodash'
import withRedux from 'next-redux-wrapper'
import Head from '../components/head'
import Form from '../components/form'
import User from '../components/user'
import Message from '../components/message'
import Navigation from '../components/navigation'
import pushesConfigs from '../config/pushes'
import { initStore } from '../reducers'
import { handleNotificationFormSubmit, handlePushesInputChange } from '../actions'
import { bindActionCreators } from 'redux'
import { get as _get } from 'lodash'
import Link from 'next/link'



class Pushes extends React.Component {

  render() {
    let { currentUser, message, hasSignup, title, formConfigs, notificationForm, ...others } = this.props,
        { username } = currentUser

    return (
      <div>
        <Head title={`Notification - ${title}`} />
        <div className="container">
          <div className="col-sm-6 col-sm-offset-3 mx-auto">

              <h1>{ title }</h1>
              <Navigation />

              <Message message={message} /> 

              <User title="Current User" user={currentUser} />
                    

              { username ? <Form form={notificationForm} { ...formConfigs } {...others} /> : 
                <p>Please <Link href="/" as="/"><a>signup</a></Link> first</p> }

          </div>
        </div>
      </div>
    )
  }
}

Pushes.propTypes = {
  title: PropTypes.string,
  message: PropTypes.string,
  currentUser: PropTypes.object,
  hasSignup: PropTypes.bool,
  onInputChange: PropTypes.func,
  onFormSubmit: PropTypes.func,
  formConfigs: PropTypes.object,
  notificationForm: PropTypes.object
}

Pushes.defaultProps = {
  title: 'Push Notification',
  formConfigs: pushesConfigs.form
}

const mapStateToProps = ({ message, hasSignup, currentUser, notificationForm }) => {
  return {
    message,
    hasSignup,
    currentUser,
    notificationForm
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onFormSubmit: bindActionCreators(handleNotificationFormSubmit, dispatch),
    onInputChange: bindActionCreators(handlePushesInputChange, dispatch)
  }
}

export { Pushes }
export default withRedux(initStore, mapStateToProps, mapDispatchToProps)(Pushes)


