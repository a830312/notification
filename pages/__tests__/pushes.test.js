import React from 'react'
import renderer from 'react-test-renderer'
import { Pushes } from '../pushes'


describe('Pushes', () => {


    it('should render with correct title', () => {
      const component = renderer.create(<Pushes title="Signup / Login" />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('should render without login / registeration form', () => {
      const component = renderer.create(<Pushes title="Signup / Login" />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('should show the form for push notification after login', () => {
      const currentUser = { username: 'abc', accessToken: '123' }
      const component = renderer.create(<Pushes title="Register" currentUser={currentUser} />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  
})

