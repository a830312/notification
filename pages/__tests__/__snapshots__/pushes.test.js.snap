// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pushes should render with correct title 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="col-sm-6 col-sm-offset-3 mx-auto"
    >
      <h1>
        Signup / Login
      </h1>
      <p>
        <a
          className="card-link"
          href="/"
          onClick={[Function]}
        >
          Register/Login
        </a>
        <a
          className="card-link"
          href="/logout"
          onClick={[Function]}
        >
          Logout
        </a>
        <a
          className="card-link"
          href="/pushes"
          onClick={[Function]}
        >
          Push Notification
        </a>
        <a
          className="card-link"
          href="/users"
          onClick={[Function]}
        >
          List users
        </a>
      </p>
      <p>
        Please 
        <a
          href="/"
          onClick={[Function]}
        >
          Register/Login
        </a>
         first
      </p>
    </div>
  </div>
</div>
`;

exports[`Pushes should render without login / registeration form 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="col-sm-6 col-sm-offset-3 mx-auto"
    >
      <h1>
        Signup / Login
      </h1>
      <p>
        <a
          className="card-link"
          href="/"
          onClick={[Function]}
        >
          Register/Login
        </a>
        <a
          className="card-link"
          href="/logout"
          onClick={[Function]}
        >
          Logout
        </a>
        <a
          className="card-link"
          href="/pushes"
          onClick={[Function]}
        >
          Push Notification
        </a>
        <a
          className="card-link"
          href="/users"
          onClick={[Function]}
        >
          List users
        </a>
      </p>
      <p>
        Please 
        <a
          href="/"
          onClick={[Function]}
        >
          Register/Login
        </a>
         first
      </p>
    </div>
  </div>
</div>
`;

exports[`Pushes should show the form for push notification after login 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="col-sm-6 col-sm-offset-3 mx-auto"
    >
      <h1>
        Register
      </h1>
      <p>
        <a
          className="card-link"
          href="/"
          onClick={[Function]}
        >
          Register/Login
        </a>
        <a
          className="card-link"
          href="/logout"
          onClick={[Function]}
        >
          Logout
        </a>
        <a
          className="card-link"
          href="/pushes"
          onClick={[Function]}
        >
          Push Notification
        </a>
        <a
          className="card-link"
          href="/users"
          onClick={[Function]}
        >
          List users
        </a>
      </p>
      <div
        className="card form-group"
      >
        <div
          className="card-body"
        >
          <div
            className="card-text"
          >
            <h4
              className="card-title"
            >
              Current User
            </h4>
            <div>
              <dt>
                username
              </dt>
              <dd>
                abc
              </dd>
            </div>
            <div>
              <dt>
                accessToken
              </dt>
              <dd>
                123
              </dd>
            </div>
          </div>
        </div>
      </div>
      <form
        action="/pushes"
        method="post"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label>
            Notification Title
          </label>
          <input
            className="form-control"
            name="title"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label>
            Notification Body
          </label>
          <input
            className="form-control"
            name="body"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="type"
            onChange={[Function]}
            type="hidden"
            value=""
          />
        </div>
        <button
          className="btn btn-warning btn-lg"
          type="submit"
        >
          Push
        </button>
      </form>
    </div>
  </div>
</div>
`;
